examples/apps/src/ble_bas_peripheral.rs

```diff
@@ -63,80 +66,125 @@ where
     )
     .unwrap();
 
-    info!("Starting advertising and GATT service");
-    let _ = join3(
-        ble_task(runner),
-        gatt_task(&server),
-        advertise_task(peripheral, &server),
-    )
-    .await;
+    let ble_runner_task = ble_task(runner);
+    let app_task = async {
```

Comment:
So is this app task here so that whenever a connection is dropped we go back to advertising? Including if there is an error for somer reason within the advertise function?

Reply:
app_task here is created so that we can run the main application loop alongside the ble_runner_task. The runner task is required to be running in the background whilst we advertise, connect, read, write, etc on the ble server, otherwise it hangs. 

In our code we've made ble_runner_task into a statically defined #[embassy::task], so it can be spawned in the background and then we can move on to the application logic. 

In this case however we can't do that because the runner is defined with a generic Controller and embassy tasks cannot contain generics.

So we have to await the ble_runner_task but also run our application logic simultaneously, and hence use that last select statement.